### ISSUE TITLE
GoFiber Example Crashes

### ISSUE THREAD

**Issue by 8BitJonny** [supertokens/supertokens-golang#199]
# What
Checkout the newest changes from #198 and make an unauthenticated call. The VerifySession Function will not catch it as an error and will call `ctx.next()` ALTHOUGH the session is unauthenticated, leading to a Nil pointer reference if you try to fetch the user session or user data later on.

# Why
Your new "fixed" implementation looks like this
```go
err = adaptor.HTTPHandlerFunc(session.VerifySession(options, func(rw http.ResponseWriter, r *http.Request) {
	c.SetUserContext(r.Context())
}))(c)
if err != nil {
	return ...
}
```
but if you look at the implementation of `adaptor.HTTPHandlerFunc` then you'll see that it is implemented like this
```go
return func(c *fiber.Ctx) error {
	handler := fasthttpadaptor.NewFastHTTPHandler(h)
	handler(c.Context())
	return nil
}
```
meaning it will always return nil, no matter the outcome of `session.VerifySession`

Go try it for yourself

**Comment by 8BitJonny** [comment]
I'm still suggesting to fix like in #195 
```go
func verifySession(options *sessmodels.VerifySessionOptions) fiber.Handler {
	return func(c *fiber.Ctx) error {
		sessionVerified := false
		_ = adaptor.HTTPHandlerFunc(session.VerifySession(options, func(rw http.ResponseWriter, r *http.Request) {
			c.SetUserContext(r.Context())
			sessionVerified = true
		}))(c)

		if !sessionVerified {
			return &fiber.Error{Message: "unauthorized"}
		}
		return c.Next()
	}
}
```

**Comment by sattvikc** [comment]
hey @8BitJonny,

Thank you for pointing this out. Going ahead with your approach with slight changes.

1. I continue to keep the error check for adaptor.HTTPHandlerFunc, just in case their (thirdparty) implementation changes in future to return any possible errors.

2. I would just return a nil error instead of fiber.Error, as returning fiber.Error changes the response to 500 and the error message provided in there, instead of what was returned by the supertokens middleware. We already have a supertokens Error handler which returns a 401 and the adaptor continues to use that response if we don't call the next on the context.

Merged PR is here - https://github.com/supertokens/supertokens-golang/pull/200
